eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java

```diff
@@ -15,25 +15,69 @@
  */
 package uk.ac.ebi.eva.accession.dbsnp.io;
 
+import com.mongodb.DuplicateKeyException;
 import org.springframework.batch.item.ItemWriter;
 import org.springframework.data.mongodb.core.MongoTemplate;
 
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpSubmittedVariantEntity;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpSubmittedVariantOperationEntity;
+import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpClusteredVariantEntity;
 import uk.ac.ebi.eva.accession.dbsnp.persistence.DbsnpVariantsWrapper;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class DbsnpVariantsWriter implements ItemWriter<DbsnpVariantsWrapper> {
 
     private MongoTemplate mongoTemplate;
 
+    private DbsnpSubmittedVariantWriter dbsnpSubmittedVariantWriter;
+
+    private DbsnpClusteredVariantWriter dbsnpClusteredVariantWriter;
+
+    private DbsnpSubmittedVariantOperationWriter dbsnpSubmittedVariantOperationWriter;
+
+
     public DbsnpVariantsWriter(MongoTemplate mongoTemplate) {
         this.mongoTemplate = mongoTemplate;
+        this.dbsnpSubmittedVariantWriter = new DbsnpSubmittedVariantWriter(mongoTemplate);
+        this.dbsnpClusteredVariantWriter = new DbsnpClusteredVariantWriter(mongoTemplate);
+        this.dbsnpSubmittedVariantOperationWriter = new DbsnpSubmittedVariantOperationWriter(mongoTemplate);
     }
 
     @Override
-    public void write(List<? extends DbsnpVariantsWrapper> items) throws Exception {
-        throw new UnsupportedOperationException("Not implemented yet. Need DbsnpClusteredVariantWriter, " +
-                                                        "DbsnpSubmittedVariantWriter and the writer for history " +
-                                                        "operations");
+    public void write(List<? extends DbsnpVariantsWrapper> wrappers) throws Exception {
+        for (DbsnpVariantsWrapper dbsnpVariantsWrapper : wrappers) {
+            List<DbsnpSubmittedVariantEntity> submittedVariants = dbsnpVariantsWrapper.getSubmittedVariants();
+            dbsnpSubmittedVariantWriter.write(submittedVariants);
+
+            List<DbsnpSubmittedVariantOperationEntity> operations = dbsnpVariantsWrapper.getOperations();
+            if (operations != null && !operations.isEmpty()) {
+                dbsnpSubmittedVariantOperationWriter.write(operations);
+            }
+        }
+        writeClusteredVariants(wrappers);
+    }
+
+    private void writeClusteredVariants(List<? extends DbsnpVariantsWrapper> items) {
+        try {
+            Collection<DbsnpClusteredVariantEntity> uniqueClusteredVariants =
+                    items.stream()
+                         .map(DbsnpVariantsWrapper::getClusteredVariant)
+                         .collect(Collectors.toMap(DbsnpClusteredVariantEntity::getHashedMessage,
```

Comment:
couldn't this be done with a simple map()? (because we are just getting the values in the next operation)